src/main/java/net/minecraftforge/common/util/ITeleporter.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2017.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.common.util;
+
+import net.minecraft.entity.Entity;
+import net.minecraft.world.World;
+import net.minecraft.world.WorldProvider;
+
+/**
+ * Interface for handling the placement of entities during dimension change.
+ *
+ * An implementation of this interface can be used to place the entity
+ * in a safe location, or generate a return portal, for instance.
+ *
+ * See the {@link net.minecraft.world.Teleporter} class, which has
+ * been patched to implement this interface, for a vanilla example.
+ */
+public interface ITeleporter
```

Comment:
I'm not quite sure what Forge's stance on this is, but I'd like to see a `@FunctionalInterface` marker here.

(Might as well use Java 8 features when it is required anyways.)

Reply:
Nope, Other single method interfaces in forge don't have this. You do not need `@FunctionalInterface` on an interface to use it as a lambda expression.